### DISCUSSION TITLE
can't create new model elements with a palette containing tool after i drag my custom Node creation from palette to diagram.

### DISCUSSION THREAD

**Discussion by Willia-Yang** [eclipse-sirius/sirius-desktop#399] [Q&A]
https://github.com/eclipse-sirius/sirius-desktop/assets/55096506/d6abb0b3-105a-4b6b-92cf-273b722ed801

**Comment by pcdavid** [comment]
Hi,

I can't see your video, but tools are not invoked by DnD but by first activating them a single click in the palette and then applying them with a second click at the target location on the diagram.

Note that if you hold the Ctrl key when applying the tool, it stays active and you can invoke it a second (and third etc.) time without re-clicking in the palette.

**Reply by Willia-Yang** [reply]
![联想截图_20240524170444](https://github.com/eclipse-sirius/sirius-desktop/assets/55096506/946462e6-7612-4b7e-a9d8-4d039722cb02)
This is the process that i did, there's no errors no warnings ,it's just gone.... so i have no idea how to solve this..
Thanks for taking time help me, really appreciate

**Comment by lredor** [comment]
Is it possible to have you VSM file (odesign file) ?
Without more details it is not possible to help you.

**Reply by Willia-Yang** [reply]
[My.zip](https://github.com/user-attachments/files/15543798/My.zip)
this is my odesign file.
Thanks for reply,

**Reply by Willia-Yang** [reply]
![tutieshi_640x388_2s](https://github.com/eclipse-sirius/sirius-desktop/assets/55096506/446d00fe-bc1a-4018-b3dd-f773f0657be6)

**Reply by lredor** [reply]
OK, so there is something strange...
The node corresponds to a `sysml::PartDefinition`
![2024-06-04_14h00_32](https://github.com/eclipse-sirius/sirius-desktop/assets/1825882/06d69e9d-b122-408b-bbbe-6475a0fc4b27)
and the tool creates a `sysml::Package`. I don't master the SysML metamodel, but I would have expected the tool to create a `sysml::PartDefinition`
![image](https://github.com/eclipse-sirius/sirius-desktop/assets/1825882/fa2be85f-193a-4995-871d-9b5613025e70)

**Reply by Willia-Yang** [reply]
I'm sorry, is a Incorrectly written. I adjusted it a bit, but it still doesn't work
![image](https://github.com/eclipse-sirius/sirius-desktop/assets/55096506/0b60a1b6-24a6-4195-9dd6-a6e094daf2b3)

**Comment by lredor** [comment]
The manipulated concept is no longer the same as before.
1- Is the "node1", that we see in your test.sysml file, well represented in your diagram?
2- When you apply the tool and save, is the semantic element created like the existing one (node1)? Indeed, I don't know if the semantic part you create is syntactically correct for Sysml.

**Reply by Willia-Yang** [reply]
1.yeah , it's well presented in our diagram
2.no, i don't think we have set the semantic part. Do you mean that we should add set process in the create tool like this?
![图片](https://github.com/eclipse-sirius/sirius-desktop/assets/55096506/7931da9e-a2d3-479a-bfbd-df0c4ac210b5)

**Comment by lredor** [comment]
I don't know. If the creation tool creates what is semantically expected, there is no reason to not have the corresponding graphical elements created.
Hence, the idea of ​​checking, after calling the creation tool (and after saving), what the contents of the sysml file look like.
